### PR TITLE
Fix RAG post openai changes

### PR DESCRIPTION
**Pull request by dusvyat**
## Description

`rag_handler`

- adjustment to core settings
- update requirements.txt, seems langchain as added more optional requirements in later version. So they are required to be installed manually
- Chroma seems to support batching under the hood now, I have removed batching

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/rag_handler/rag.py [hunk 1]
```diff
@@ -100,6 +102,8 @@ def extract_generated_text(response: str) -> str:
         try:
             if "choices" in data:
                 return data["choices"][0]["text"]
+            elif isinstance(data, Completion):
+                return data.choices[0].text
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/rag.py:106]: for new openai api, forgot to make the change previously

File: mindsdb/integrations/handlers/rag_handler/settings.py [hunk 2]
```diff
@@ -51,8 +51,8 @@
 {question}
 """
 
-DEFAULT_CHUNK_SIZE = 500
-DEFAULT_CHUNK_OVERLAP = 50
+DEFAULT_CHUNK_SIZE = 750
+DEFAULT_CHUNK_OVERLAP = 250
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/settings.py:55]: these are more optimal defaults

File: mindsdb/integrations/handlers/rag_handler/settings.py [hunk 3]
```diff
@@ -106,9 +106,7 @@ def get_available_writer_model_ids(args: dict) -> list:
 def get_available_openai_model_ids(args: dict) -> list:
     """Get available openai LLM model ids"""
 
-    openai.api_key = args["openai_api_key"]
-
-    models = openai.OpenAI().models.list().data
+    models = openai.OpenAI(api_key=args["openai_api_key"]).models.list().data
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/settings.py:109]: before it was using env variable as we needed to update for new openai interface

File: mindsdb/integrations/handlers/rag_handler/settings.py [hunk 4]
```diff
@@ -280,7 +278,7 @@ class RAGBaseParameters(BaseModel):
     llm_params: Any
     vector_store_folder_name: str
     use_gpu: bool = False
-    embeddings_batch_size: int = MAX_EMBEDDINGS_BATCH_SIZE
+    embeddings_batch_size: int = MAX_EMBEDDINGS_BATCH_SIZE  # not used, leaving in place to prevent breaking changes
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/settings.py:281]: leaving for now to allow for backward compatibility when loading users previously created models. We can remove at a later date

File: mindsdb/integrations/handlers/rag_handler/rag.py [hunk 5]
```diff
@@ -2,6 +2,8 @@
 from collections import defaultdict
 from typing import List
 
+from openai.types import Completion
```
[the remark this answers is not in the document]
Comment (dusvyat) [reply mindsdb/integrations/handlers/rag_handler/rag.py:5]: yes type is still valid. See [here](https://github.com/openai/openai-python/blob/main/src/openai/types/completion.py)

File: mindsdb/integrations/handlers/rag_handler/settings.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dusvyat) [reply mindsdb/integrations/handlers/rag_handler/settings.py:109]: good idea, let me investigate. no harm in adding in

File: mindsdb/integrations/handlers/rag_handler/settings.py [hunk 6]
```diff
@@ -106,9 +106,7 @@ def get_available_writer_model_ids(args: dict) -> list:
 def get_available_openai_model_ids(args: dict) -> list:
     """Get available openai LLM model ids"""
 
-    openai.api_key = args["openai_api_key"]
-
-    models = openai.OpenAI().models.list().data
+    models = openai.OpenAI(api_key=args["openai_api_key"], base_url=args["base_url"]).models.list().data
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/settings.py:110]: adding in `base_url` support for openai

File: mindsdb/integrations/handlers/rag_handler/settings.py [hunk 7]
```diff
@@ -266,9 +264,11 @@ def load_writer_llm(self) -> Writer:
 
     def load_openai_llm(self) -> partial:
         """Load OpenAI LLM API interface"""
-        client = openai.OpenAI(api_key=self.config_dict["openai_api_key"])
+        client = openai.OpenAI(api_key=self.config_dict["openai_api_key"], base_url=self.config_dict["base_url"])
         config = self.config_dict.copy()
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/settings.py:268]: adding in `base_url` support for openai

**Review by paxcema — APPROVED** [review]
LGTM

**Pull request merged**